compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -297,16 +298,32 @@ private void replaceMockedFunctions(BIRPackage birPkg, Map<String, String> mockF
             for (BIRBasicBlock functionBasicBlock : functionBasicBlocks) {
                 BIRTerminator bbTerminator = functionBasicBlock.terminator;
                 if (bbTerminator.kind.equals(InstructionKind.CALL)) {
-                    //We get the callee and the name and generate 'calleepackage#name'
                     BIRTerminator.Call callTerminator = (BIRTerminator.Call) bbTerminator;
-                    String functionKey = callTerminator.calleePkg.toString() + MOCK_ANNOTATION_DELIMITER
-                            + callTerminator.name.toString();
+                    String functionKey = null;
+
+                    // Generate the function key
+                    if (callTerminator.isVirtual && callTerminator.args.size() == 1) {
```

Comment:
Do we need this second check? What if the object-method call has more than one argument?

Reply:
The argument considered in this case is the object reference we are using to call the object function. ie. p1.getName() for an object Person p1, the argument, in this case, is p1. 

I can remove the second check for size, but there are other virtual functions like `__init` that would get caught in this block then. When I add the size check, only the object methods are considered.

Reply:
You are correct. The size check is not working when arguments are considered

Reply:
Removed the check for arguments. It looks only for isVirtual